src/napari_micromanager/_mda_handler.py

```diff
@@ -150,11 +170,21 @@ def _on_mda_frame(self, image: np.ndarray, event: MDAEvent) -> None:
 
         # move the viewer step to the most recently added image
         # this seems to work better than self.viewer.dims.set_point(a, v)
-        cs = list(self.viewer.dims.current_step)
-        for a, v in enumerate(im_idx):
-            cs[a] = v
-        self.viewer.dims.current_step = tuple(cs)
 
+        if im_idx > self._largest_idx:
+            self._largest_idx = im_idx
+            # Processing the most recent event
+            # update the viewer in the main thread
+            cs = list(self.viewer.dims.current_step)
+            for a, v in enumerate(im_idx):
+                cs[a] = v
+            self._update_viewer_dims(cs, layer_name, event)
```

Comment:
slightly wary of accessing `self.viewer` from a thread.  might want some locks in here.  Also, I wonder it would be possible for the thing triggering `_update_viewer_dims` to not be `_process_frame`

Reply:
How about just moving that access in to the function that has `ensure_main_thread`?

Reply:
sure, and then it would just be `self._update_viewer_dims(layer_name, event)`?  if that works too, I like that